server/applicationPattern/applicationPattern/onfModel/models/layerProtocols/TcpClientInterface.js

```diff
@@ -142,13 +148,31 @@ class TcpClientInterface extends layerProtocol {
                 let tcpClientPac = layerProtocol[onfAttributes.LAYER_PROTOCOL.TCP_CLIENT_INTERFACE_PAC];
                 let tcpClientConfiguration = tcpClientPac[onfAttributes.TCP_CLIENT.CONFIGURATION];
                 remoteProtocol = tcpClientConfiguration[onfAttributes.TCP_CLIENT.REMOTE_PROTOCOL];
+                remoteProtocol = (await TcpClientInterface.getProtocolFromProtocolEnum(remoteProtocol))[0];
```

Comment:
This only has to be here, because BasicServices has hardcoded 'HTTPS' instead of using the enum. As I wrote before, it's a lot less confusing when we would use just the "tcp-client-interface-1-0:PROTOCOL_TYPE_HTTPS" everywhere and only change it to simple form, when it's needed. If you change basic services, none of these transformations are actually needed.

Reply:
getRemoteProtocolAsync function is used in various places . Where in majority of the places it is used for below mentioned functionalities  

- For comparing it with the incoming values of request body 
- For formulating the request body for forwardings to be sent

 and in both the places only the actual value of  protocol is expected and not the onf notation . 

So , thought it would be more effective to declare it here.